command/agent/config_test.go

```diff
@@ -1068,3 +1071,109 @@ func TestReadConfigPaths_dir(t *testing.T) {
 		t.Fatalf("bad: %#v", config)
 	}
 }
+
+func TestUnixSockets(t *testing.T) {
```

Comment:
There are already two helper functions there; the rest just toggles various combinations of bad data. I think rewiring it in the way you are suggesting will not actually make it any shorter.